objects/src/block/note_tree.rs

```diff
@@ -0,0 +1,53 @@
+use miden_crypto::{
+    hash::rpo::RpoDigest,
+    merkle::{MerkleError, SimpleSmt},
+};
+
+use crate::{notes::NoteMetadata, BLOCK_OUTPUT_NOTES_TREE_DEPTH, MAX_NOTES_PER_BATCH};
+
+/// Wrapper over [SimpleSmt] for notes tree
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub struct BlockNoteTree(SimpleSmt<BLOCK_OUTPUT_NOTES_TREE_DEPTH>);
+
+impl BlockNoteTree {
+    /// Returns a new [BlockOutputNotesTree].
+    ///
+    /// All leaves in the returned tree are set to [ZERO; 4].
+    pub fn new() -> Self {
+        Self(SimpleSmt::new().expect("Unreachable"))
+    }
+
+    /// Returns a new [BlockOutputNotesTree] instantiated with entries set as specified by the provided entries.
+    ///
+    /// Entry format: (batch_index, note_index, (note_id, note_metadata)).
+    ///
+    /// All leaves omitted from the entries list are set to [ZERO; 4].
+    ///
+    /// # Errors
+    /// Returns an error if:
+    /// - The number of entries exceeds the maximum notes tree capacity, that is 2^21.
+    /// - The provided entries contain multiple values for the same key.
+    pub fn with_entries(
```

Comment:
The comments need to be updated not to refer to `BlockOutputNotesTree`. Also, where do we use this method in the node?

Reply:
> Also, where do we use this method in the node?

We use it here: https://github.com/0xPolygonMiden/miden-node/blob/bacb3da4e348f0481e026bdeb02c973337b3e368/store/src/state.rs#L492